### PR TITLE
feat(dashbaords): add dashboard favourite to table view

### PR DESCRIPTION
**Pull request by harshithadurai**
Add dashboard favourite column to table view:

Screenshot 2024-12-06 at 12 11 35 PM

Issue: #78023

### CODE REVIEW

File: static/app/views/dashboards/manage/dashboardTable.tsx [hunk 1]
```diff
@@ -232,6 +293,20 @@ function DashboardTable({
...
       grid={{
         renderBodyCell,
+        renderHeadCell: column => {
+          if (column.key === ResponseKeys.FAVORITE) {
+            return (
+              <Feature features="dashboards-favourite">
+                <StyledIconStar
+                  color="yellow300"
+                  isSolid
+                  aria-label={t('Favorite Header')}
```
Comment (nikkikapadia) [inline static/app/views/dashboards/manage/dashboardTable.tsx:303]:
```suggestion
                  aria-label={t('Favorites Column')}
```

File: static/app/views/dashboards/manage/dashboardTable.tsx [hunk 2]
```diff
@@ -50,6 +53,55 @@ enum ResponseKeys {
...
+  api,
+  organization,
+  dashboardId,
+}: FavoriteButtonProps) {
+  const [favorited, setFavorited] = useState(isFavorited);
+  return (
+    <Feature features="dashboards-favourite">
+      <StyledFavoriteButton
+        aria-label={t('Favorite Button')}
+        data-test-id="xx"
```
Comment (nikkikapadia) [inline static/app/views/dashboards/manage/dashboardTable.tsx:77]: if this isn't being used can we take this out? or make it more comprehensive?

**Review by nikkikapadia — APPROVED** [review]
yayy looks good! just a couple smol things

File: static/app/views/dashboards/manage/dashboardTable.tsx (hunk 2 above)
Comment (harshithadurai) [reply static/app/views/dashboards/manage/dashboardTable.tsx:77]: oh oops I was using it for testing something

**Pull request merged**